src/ast/passes/codegen_llvm.cpp

```diff
@@ -1154,15 +1154,29 @@ void CodegenLLVM::visit(Call &call)
                            "cat",
                            AsyncAction::cat);
   } else if (call.func == "exit") {
-    /*
-     * perf event output has: uint64_t asyncaction_id
-     * The asyncaction_id informs user-space that this is not a printf(), but is
-     * a special asynchronous action. The ID maps to exit().
-     */
-    AllocaInst *perfdata = b_.CreateAllocaBPF(b_.getInt64Ty(), "perfdata");
-    b_.CreateStore(b_.getInt64(asyncactionint(AsyncAction::exit)), perfdata);
-    b_.CreateOutput(ctx_, perfdata, sizeof(uint64_t), &call.loc);
-    b_.CreateLifetimeEnd(perfdata);
+    auto elements = AsyncEvent::Exit().asLLVMType(b_);
+    StructType *exit_struct = b_.GetStructType("exit_t", elements, true);
+    AllocaInst *buf = b_.CreateAllocaBPF(exit_struct, "exit");
+    size_t struct_size = datalayout().getTypeAllocSize(exit_struct);
+
+    // Fill in perf event struct.
+    b_.CreateStore(
+        b_.getInt64(asyncactionint(AsyncAction::exit)),
+        b_.CreateGEP(exit_struct, buf, { b_.getInt64(0), b_.getInt32(0) }));
+    auto code = 0;
+    if (call.vargs.size() == 1) {
+      auto code_opt = bpftrace_.get_int_literal(call.vargs.at(0));
```

Comment:
so can this not be a variable?

Reply:
this will return an `std::optional` which should be checked in the latter line. we can't directly assign it to the `code` variable.

Reply:
I think what @jordalgo meant was a script using a variable like this:
```
$exit_code = 1;
exit($exit_code);
```

We should be able to make this work. `get_int_literal` only lets us get the value of a literal integer, but if you call `accept(call.vargs.at(0))` instead, that that'll recurse the AST and populate `expr_` with an expression representing either a literal or a non-literal value. Take a look at the implementation of the `signal` function for a real example.